text/0002-searchable-display-attributes-behaviors.md

```diff
@@ -0,0 +1,123 @@
+
+### Summary
+
+Ever since the beginning, the design of `searchableAttributes` and `displayedAttributes` as shown flaws. The introduction of the wildcard tried to adress some of those flaws, but there are still issues that this specs tries to correct.
+
+Searchable attributes have several uses:
+- know in which fields we can search.
+- know the importance of fields for the criterion `Attributes`.
+
+The main issues we want to adress here are:
+- The question of `searchableAttributes` order when the wildcard is used.
+- The fact that already known attributes (e.g from facets), are not indexed correctly is put before the documents (bug), need to be taken into account in the design.
+
+Is it relevant to mix this 2 information when we have wildcards?
+
+### Explanation
+
+#### default values
+
+`searchable_attributes`: *   
+`displayed_attributes`: *
+
+#### wildcard (*) & `searchable_attributes` order
+
+> When the searchable attributes value is set to ["*"] the priority of the documents attributes is undefined.
+
+By default searchable attributes should be in the order of the first indexed document, for example:
+```json
+{
+  "title": "many the fish",
+  "description": "it is awfull to be liked by many"
+}
+```
+default `searchable_attributes` order should be `["title", "description"]` where `"title" > "description"`
+
+When a wildcard is set we should have the default behavior.
+
+#### `searchable_attributes` & `displayed_attributes`
+
+what is the behavior for a field that:
+- > **is in `searchable_attributes` and in `displayed_attributes`:**
+  > meilisearch will use the field for a search and the field will be returned in search results
+- > **is in `searchable_attributes` but not in `displayed_attributes`:**
+  > meilisearch will use the field for a search but the field will **not** be returned in search results
+- > **is not in `searchable_attributes` but in `displayed_attributes`:**
+  > meilisearch will **not** use the field for a search but the field will be returned in search results
+- > **is not in `searchable_attributes` and not in `displayed_attributes`:**
+  > meilisearch will **not** use the field for a search and the field will **not** be returned in search results. The field is unused but should be kept in case of it is added in `searchable_attributes` or `displayed_attributes`.
+
+#### add a new configuration `attributes_order`?
+
+This new config may define the order of attributes for the criterion `Attributes`, no wildcard allowed.
+if `searchable_attribute` is: `["title", "desc"]`
+and `attributes_order` is: `["desc", "id", "title"]`
+The searchable attributes order would be:
+`"desc" > "title"` (`"id"` is not searchable so is ignored by the criterion `Attributes`).
+
+`attributes_order` is always exhaustive, If a new field is added, it is "pushed" at the end of the list.
+
+##### drawbacks:
+- new config added
+- more complexity
+- could be replace by a small note about wildcard in `searchable_attributes`
```

Comment:
```suggestion
- could be replaced by a small note about wildcard in `searchable_attributes`
```

Reply:
This idea needs some development, it isn't clear at all what it means to replace this with a 'small note' :)

Reply:
domumenting strange behaviour != fixing strange behaviour, this is a non-solution to me. @bidoubiwa will tell you that people don't really read the doc in depth the first time around. What we want is discoverability + intuitiveness

Reply:
I think you tend to read the doc about a specific subject if you think you don't understand it. In the case of `searchableAttribute` the meaning is very clear in its first use `in what attributes will meilisearch search`. Because that meaning is so intuitive, you wouldn't guess that there is a second one to look after in the documentation.

So even if they in fact read the documentation, they would probably just read over that part because they think they already understand it. 

This is speculation of course, maybe this is not how users read. This is how I read and how I think people read.

Reply:
Oh, men, I am in your team now, @MarinPostma and @bidoubiwa !